src/main/java/org/mule/service/oauth/internal/DefaultAuthorizationCodeOAuthDancer.java

```diff
@@ -349,14 +354,15 @@ private static void sendResponse(StateDecoder stateDecoder, HttpResponseReadyCal
     }
   }
 
-  private static void sendResponse(HttpResponseReadyCallback responseCallback, HttpStatus status, String message,
-                                   String locationHeader) {
+  private void sendResponse(HttpResponseReadyCallback responseCallback, HttpStatus status, String message,
+                            String locationHeader) {
     HttpResponseBuilder httpResponseBuilder = HttpResponse.builder();
     httpResponseBuilder.statusCode(status.getStatusCode());
     httpResponseBuilder.reasonPhrase(status.getReasonPhrase());
     httpResponseBuilder.entity(new ByteArrayHttpEntity(message.getBytes()));
     httpResponseBuilder.addHeader(CONTENT_LENGTH, valueOf(message.length()));
     httpResponseBuilder.addHeader(LOCATION, locationHeader);
+    httpResponseBuilder.headers(new MultiMap<>(customHeaders.get()));
```

Comment:
use `MultiMap. toImmutableMultiMap()`

Reply:
this multimap is ephemeral. Nobody will get the chance to try an mutate it. Adding the immutable convertion will lead to gainless perf overhead